src/runtime/transaction_execution.zig

```diff
@@ -180,6 +180,7 @@ pub const CopiedAccount = struct {
 
 pub const ExecutedTransaction = struct {
     err: ?InstructionErrorEnum,
+    err_index: ?u32,
```

Comment:
I don't think these can be independently null or non-null, so I would suggest tying that state together: `err: ?struct { index: u32, value: InstructionErrorEnum }`.